src/plugins/intel_npu/tests/functional/internal/compiler_adapter/zero_graph.hpp

```diff
@@ -0,0 +1,249 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <gtest/gtest.h>
+#include <stdlib.h>
+
+#include <common_test_utils/test_assertions.hpp>
+
+#include "common/npu_test_env_cfg.hpp"
+#include "common_test_utils/subgraph_builders/multi_single_conv.hpp"
+#include "common_test_utils/test_constants.hpp"
+#include "intel_npu/utils/utils.hpp"
+#include "intel_npu/utils/zero/zero_mem.hpp"
+#include "intel_npu/utils/zero/zero_utils.hpp"
+#include "ir_serializer.hpp"
+#include "ze_graph_ext_wrappers.hpp"
+#include "zero_init_mock.hpp"
+
+using namespace intel_npu;
+
+namespace {
+size_t calculate_size_with_alignment_padding(size_t size, size_t alignment) {
+    return size + alignment - (size % alignment);
+}
+
+size_t get_file_size(std::ifstream& file) {
+    auto size = file.tellg();
+    file.seekg(0, std::ios::end);
+    size = file.tellg() - size;
+    file.seekg(0, std::ios::beg);
+
+    return size;
+}
+}  // namespace
+
+namespace ov::test::behavior {
+class ZeroGraphTest : public ::testing::TestWithParam<std::tuple<int, int>> {
+public:
+    std::shared_ptr<ZeroInitStructsHolder> zeroInitStruct;
+
+    std::shared_ptr<ZeGraphExtWrappers> zeGraphExt;
+
+    SerializedIR serializedIR;
+
+    GraphDescriptor graphDescriptor;
+
+    std::shared_ptr<ov::Model> model;
+
+    std::shared_ptr<driver_compiler_utils::IRSerializer> irSerializer;
+
+    std::string blobPath;
+
+    int extVersion;
+
+    int graphDescFlag;
+
+    static std::string getTestCaseName(testing::TestParamInfo<std::tuple<int, int>> obj) {
```

Comment:
```suggestion
    static std::string getTestCaseName(const testing::TestParamInfo<std::tuple<int, int>>& obj) {
```

PR https://github.com/openvinotoolkit/openvino/pull/31698 optimized all these params, let's not bring regressions.